ride/swap.ride

```diff
@@ -0,0 +1,277 @@
+      []
+    ).exactAs[String]
+    (isReverse, pool)
+  } else if (lpReverse != unit) then {
+    let isReverse = true
+    let pool = invoke(
+      factoryContract,
+      "getPoolAddressFromLpAssetREADONLY",
+      [lpReverse],
+      []
+    ).exactAs[String]
+    (isReverse, pool)
+  } else {
+    poolNotExist
+  }
+
+  let poolStatus = invoke(
+    factoryContract,
+    "getPoolStatusREADONLY", 
+    [pool], 
+    []
+  ).exactAs[Int]
+  strict check = poolStatus == 1 || poolDisabled
+  let poolAdr = addressFromStringValue(pool.valueOrElse(""))
+
+  let res = invoke(
+    poolAdr,
+    "calculateAmountOutForSwapREADONLY",
+    [cleanAmountIn.toInt(), isReverse],
+    []
+  ).exactAs[Int]
+
+
+  (nil,
+  res)
+}
+
+@Callable(i)
+func swap(
+  amountOutMin: Int,
+  assetOut: String,
+  addressTo: String 
+) = {
+  let pmt = i.payments[0].value()
+  let assetIn = if(pmt.assetId == unit) then "WAVES" else pmt.assetId.value().toBase58String()
+  let amountIn = pmt.amount.value()
+  let prFee = getInteger(protocolFee()).value()
+  let feeProtocolAmountCalc = fraction(amountIn.toBigInt(), prFee.toBigInt(), feeScale)
+  let plFee = getInteger(poolFee()).value()
+  let feePoolAmountCalc = fraction(amountIn.toBigInt(), plFee.toBigInt(), feeScale)
+
+  let feeProtocolAmount = if (feeProtocolAmountCalc.toInt() == 0) then {
+    toBigInt(1)
+  } else {
+    feeProtocolAmountCalc
+  }
+  let feePoolAmount = if (feePoolAmountCalc.toInt() == 0) then {
+    toBigInt(1)
+  } else {
+    feePoolAmountCalc
+  }
+  let cleanAmountIn = amountIn.toBigInt() - feeProtocolAmount - feePoolAmountCalc
+
+  strict checks = [
+    i.payments.size() == 1 || "exactly 1 payment are expected".throw()
+  ]
+
+  let lpNonReverse = invoke(
+    factoryContract,
+    "getLpAssetFromPoolAssetsREADONLY",
+    [assetIn, assetOut],
+    []
+  ).as[String]
+
+  let lpReverse = invoke(
+    factoryContract,
+    "getLpAssetFromPoolAssetsREADONLY",
+    [assetOut, assetIn],
+    []
+  ).as[String]
+  let (isReverse, pool) = if (lpNonReverse != unit) then {
+    let isReverse = false
+    let pool = invoke(
+      factoryContract,
+      "getPoolAddressFromLpAssetREADONLY",
+      [lpNonReverse.value()],
+      []
+    ).exactAs[String]
+    (isReverse, pool)
+  } else if (lpReverse != unit) then {
+    let isReverse = true
+    let pool = invoke(
+      factoryContract,
+      "getPoolAddressFromLpAssetREADONLY",
+      [lpReverse.value()],
+      []
+    ).exactAs[String]
+    (isReverse, pool)
+  } else {
+    # poolNotExist
+    ("pool is not exist: assetIn=" + assetIn + " assetOut=" + assetOut + " lpReverse=" + if (lpReverse == unit) then "unit" else lpReverse.value() + " lpNonReverse=" + if (lpNonReverse == unit) then "unit" else lpNonReverse.value() + "").throw()
+  }
+  
+  let poolStatus = invoke(
+    factoryContract,
+    "getPoolStatusREADONLY", 
+    [pool], 
+    []
+  ).exactAs[Int]
+  strict check = poolStatus == 1 || poolDisabled # ACTIVE, pool without restrictions
```

Comment:
Is it enough or flexible solution to pause swaps?

At least we already have 1 thing unused here: `isPoolOneTokenOperationsDisabledREADONLY`

Reply:
Yes, it's flexible
isPoolOneTokenOperationsDisabledREADONLY - it's not just for swap. it is also for putting liquidity. And not usable in swap